src/main/java/org/elasticsearch/cluster/service/InternalClusterService.java

```diff
@@ -523,6 +538,12 @@ public void run() {
         }
     }
 
+    private void warnAboutSlowTaskIfNeeded(long executionTime, String source) {
+        if (executionTime >= slowTaskLoggingThreshold.getMillis()) {
+            logger.warn("cluster state update task [{}] took too long: {}ms", source, executionTime);
```

Comment:
@kimchy it's 30 seconds by default and can be changed dynamically. So we can either 1) increase the default value to something that we feel comfortable with or 2) make it logged on info level. Not sure if the second option makes a lot of sense since we are already logging timing on the debug level. So, I thought the whole idea to log long running tasks was to "warn" users about a potential issue.